### PR TITLE
Add setDiag support for any square domain

### PR DESCRIPTION
**Pull request by wongalvis14**
setDiag now supports any regular square domain, as long as it is in the form of {a..#n, b..#n}

### CODE REVIEW

**Comment by e-kayrakli** [comment]
Thanks @wongalvis14!

> as long as it is in the form of {a..#n, a..#n}

Is there a major roadblock for relaxing this? It sounds it would be easy to support {a..#n, b..#m}?

**Comment by wongalvis14** [comment]
@e-kayrakli Yes, I didn't figure out right away yesterday. I've pushed the update that supports {a..#n, b..#n}. Having rectangular domains is a bit trickier.

File: test/library/packages/LinearAlgebra/performance/diag/setdiag-test.chpl [hunk 1]
```diff
@@ -2,7 +2,7 @@ use LinearAlgebra;
 
 config const m=10;
 
-const Space = {1..m, 1..m};
+const Space = {42..#m, 24..#m};
```
Comment (e-kayrakli) [inline test/library/packages/LinearAlgebra/performance/diag/setdiag-test.chpl]: Can you keep the old case and this as a separate case? Note that it doesn't need to be in a separate test. Just expand this test to check:

- 1-based domains
- 0-based domains
- negative-based domains
- something random like what you have here

**Comment by e-kayrakli** [comment]
@wongalvis14 -- I am closing this PR because it has been very active. Please feel free to discuss here about any questions you may have and reopen if you make some progress on it. Thanks!

**Pull request closed without merging**